### PR TITLE
DM-44699: Add Group Details page to the CM web app

### PR DESCRIPTION
**Pull request by emanehab99**
Add group details page to CM monitoring web app
screenshot-0 0 0 0_8080-2024 06 25-16_06_53

### CODE REVIEW

**Comment by eacharles** [comment]
Overall, this looks good.  There is, however, one subtle issue that we might have to come back to at some point, in that there is overlap between the wms_task_reports from the various jobs in a group, as the later jobs represent attempts to reprocess failed jobs from the earlier reports, so we might, at some point, decide that summing up all the wms_reports insn't what we want.  However, that is an issue for a later day, and for now I think we should just merge this.

**Review by eacharles — APPROVED** [review]

**Comment by emanehab99** [comment]
> Overall, this looks good. There is, however, one subtle issue that we might have to come back to at some point, in that there is overlap between the wms_task_reports from the various jobs in a group, as the later jobs represent attempts to reprocess failed jobs from the earlier reports, so we might, at some point, decide that summing up all the wms_reports insn't what we want. However, that is an issue for a later day, and for now I think we should just merge this.

Thanks for the feedback @eacharles !
I cannot merge it though. All pages are now broken after I rebased my branch. Looks like there were many changes to the wrappers and db classes. I need to figure out the issue first

**Comment by eacharles** [comment]
Hmm.  I’ll have a look when I get to SLAC.-eOn Jun 26, 2024, at 12:13 AM, Eman Ali ***@***.***> wrote:﻿

Overall, this looks good. There is, however, one subtle issue that we might have to come back to at some point, in that there is overlap between the wms_task_reports from the various jobs in a group, as the later jobs represent attempts to reprocess failed jobs from the earlier reports, so we might, at some point, decide that summing up all the wms_reports insn't what we want. However, that is an issue for a later day, and for now I think we should just merge this.

Thanks for the feedback @eacharles !
I cannot merge it though. All pages are now broken after I rebased my branch. Looks like there were many changes to the wrappers and db classes. I need to figure out the issue first

File: src/lsst/cmservice/web_app/templates/group_details.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {% extends "base.html" %}
 {% block title %} Group Details {% endblock %}
 {% block content %}
-<nav class="flex px-2 sm:px-4 lg:px-4" aria-label="Breadcrumb">
+<nav class="flex px-2 sm:px-4 lg:px-4" aria-label="Breadcrumb" xmlns="http://www.w3.org/1999/html">
```
Comment (mr-asher) [inline src/lsst/cmservice/web_app/templates/group_details.html:4]: why are you explicitly adding the xmlns attribute? This seems weird for a nav bar?

File: src/lsst/cmservice/web_app/templates/group_details.html [hunk 2]
```diff
@@ -27,10 +27,146 @@
   </ol>
 </nav>
 <div class="w-full">
-    <div class="overflow-hidden px-4 py-5 sm:p-6 relative w-1/4" xmlns="http://www.w3.org/1999/html">
+    <div class="overflow-hidden px-4 py-5 sm:p-6 relative w-1/2" xmlns="http://www.w3.org/1999/html">
```
Comment (mr-asher) [inline src/lsst/cmservice/web_app/templates/group_details.html:30]: And here again? If this is required maybe add it to the html tag in base?

**Review by mr-asher — APPROVED** [review]
This looks great to me. Nice work.

File: src/lsst/cmservice/web_app/templates/group_details.html (hunk 2 above)
Comment (emanehab99) [reply src/lsst/cmservice/web_app/templates/group_details.html]: Not required. removed!

File: src/lsst/cmservice/web_app/templates/group_details.html (hunk 1 above)
Comment (emanehab99) [reply src/lsst/cmservice/web_app/templates/group_details.html]: Removed

**Pull request merged**